### PR TITLE
Add custom applicaton support (work in progress)

### PR DESCRIPTION
**Pull request by mvantellingen**
Example usage::

```go
newClient.CustomApplicationGetWithID(context.TODO(), "ckcmb5qvmzpes0714901hx27v")

draft := &commercetools.CustomApplicationDraft{
	Name: "terraform-test",
	NavbarMenu: commercetools.CustomApplicationNavbarMenu{
		Key:         "foobar",
		URIPath:     "no-idea",
		Permissions: []string{"ViewCategories"},
		LabelAllLocales: []commercetools.CustomApplicationLabelLocale{
			commercetools.CustomApplicationLabelLocale{
				Locale: "NL",
				Value:  "Michael Test",
			},
		},
		Submenu: []commercetools.CustomApplicationNavbarSubmenu{},
	},
}

app, err := newClient.CustomApplicationCreate(context.TODO(), draft)
if err != nil {
	panic(err)
}

```

### CODE REVIEW

**Comment by mvantellingen** [comment]
@emmenko which permissions do we need to actually create a custom application. Retrieving works fine but when creating a new one I receive the following
```json
{"errors":[{"message":"Missing or invalid authentication token.","locations":[{"line":3,"column":4}],"path":["createProjectExtensionApplication"],"extensions":{"code":"UNAUTHENTICATED"}}],"data":{"createProjectExtensionApplication":null}}
```

**Comment by emmenko** [comment]
Hi @mvantellingen , thanks for taking over 🙏

To manage custom applications, you need an access token with the scope `manage_project_settings`. For retrieving a custom application, you simply need an access token with access to the project (could be any scope).

Let me know if you need any more assist. I'm also happy to review the PR if you need me to.
Thanks

**Comment by emmenko** [comment]
FYI: we're planning to change some things about how you manage and register Custom Applications. The current API should stay for backwards compatibility, but there would be a new one.

I'll get in touch once we have it in place.

File: commercetools/service_custom_applications.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+				}
+			}
+		}`)
+
+	query.Bind("draft", draft)
+	err := query.Execute(&result, query.ForMerchantCenter())
+	if err != nil {
+		return nil, err
+	}
+	// if len(result.CreateProjectExtensionApplication.Applications) < 1 {
```
Comment (mvantellingen) [inline commercetools/service_custom_applications.go:95]: @emmenko I'm looking at how we can get the created custom app back via graphql. Any ideas? I thought as a workaround to sort the applications object based on last created (although that is also not really safe)

**Comment by mvantellingen** [comment]
> FYI: we're planning to change some things about how you manage and register Custom Applications. The current API should stay for backwards compatibility, but there would be a new one.
> 
> I'll get in touch once we have it in place.

Hi @emmenko,

I would like to take some steps to get the support in the terraform provider (we really need it for a couple of components we are creating). Is the new API already available?

**Comment by demeyerthom** [comment]
Custom apps can only be configured through the GraphQL instead

**Pull request closed without merging**